lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/PointCloud.java

```diff
@@ -0,0 +1,298 @@
+        if (relativeEndPointsWithHit == null) {
+            relativeEndPointsWithHit = pointsReference.toRelative.transform(this, POINTS_WITH_HITS);
+        }
+        return relativeEndPointsWithHit;
+    }
+
+    /**
+     * Returns all end points of all rays of this {@link PointCloud} in absolute coordinates.
+     * Absolute coordinates represent world coordinates, and have {@link #getOrigin()} and {@link #getOrientation()}
+     * already applied.
+     *
+     * @return a list of points in absolute coordinates
+     */
+    public List<Point> getAbsoluteEndPoints() {
+        if (absoluteEndPoints == null) {
+            absoluteEndPoints = pointsReference.toAbsolute.transform(this, POINTS_ALL);
+        }
+        return absoluteEndPoints;
+    }
+
+    /**
+     * Returns all end points of all rays of this {@link PointCloud} that hit something in absolute coordinates.
+     * Absolute coordinates represent world coordinates, and have {@link #getOrigin()} and {@link #getOrientation()}
+     * already applied.
+     *
+     * @return a list of points in absolute coordinates
+     */
+    public List<Point> getAbsoluteEndPointsWithHit() {
+        if (absoluteEndPointsWithHit == null) {
+            absoluteEndPointsWithHit = pointsReference.toAbsolute.transform(this, POINTS_WITH_HITS);
+        }
+        return absoluteEndPointsWithHit;
+    }
+
+
+    /**
+     * Returns the type of reference frame the points are stored in this point cloud internally. The usage of
+     * {@link #getAbsoluteEndPoints()} and {@link #getRelativeEndPoints()} is independent of the value returned
+     * by this method, as transformation is already done internally if necessary. Therefore, it should
+     * usually not be required to use this method, except for (de)serialization use-cases.
+     *
+     * @return the {@link PointReference} type of the points stored in the point cloud.
+     */
+    public PointReference getReferenceFormat() {
+        return pointsReference;
+    }
+
+    /**
+     * A {@link Point} of the point cloud consists of its coordinates, an identifier
+     * of the type of object the point has hit, and the distance to the point cloud origin.
+     */
+    public static class Point extends Vector3d {
+
+        private static final long serialVersionUID = 1L;
+
+        private final byte hitType;
+        private final float distance;
+
+        /**
+         * @param endPoint the coordinates of the point cloud
+         * @param distance the distance to the origin of the point cloud
+         * @param hitType  the type of hit object represented by this point. 0 = no hit
+         */
+        public Point(Vector3d endPoint, float distance, byte hitType) {
+            x = endPoint.x;
+            y = endPoint.y;
+            z = endPoint.z;
+            this.distance = distance;
+            this.hitType = hitType;
+        }
+
+        /**
+         * @return true if the ray generating this {@link Point} has hit an object
```

Comment:
These comments will still result in a checkstyle warning 🤷

Reply:
tbh, then we should update our checkstyle.xml. My solution would be to either do:

```
 /**
    * Returns true if the ray generating this {@link Point} has hit an object.
    */
```

```
 /**
    * Returns true if the ray generating this {@link Point} has hit an object
    * @return true if the ray generating this {@link Point} has hit an object
    */
```

.. or I could remove the comment completely. Those three solutions would please checkstyle, but would not help us at all.

Reply:
At least for simple getters/setters we should allow a simplified comment.

Reply:
Yes, I agree. From a rendering perspective there's also not much of a difference (at least for IntelliJ):

__@return:__
![grafik](https://github.com/user-attachments/assets/78704b98-a639-4dcf-b722-40fa6220d11d)

__Returns:__
![grafik](https://github.com/user-attachments/assets/d991d26f-38c6-4fb5-962f-29b1032fc3c6)

When reading the code, I personally prefer the fields themselves to be well commented. However, this isn't very useful when working with the compiled code where you would only have access to the comments of the API-methods (?!).

Anyhow, we shouldn't just increase the number of checkstyle warnings, but rather include exception policies.

Reply:
At least from what I've found there is no possibility to allow missing description, and you can also not have different rules for the comment style for getters/setters and other methods (you can only define, that public getters/setters are allowed to have no javadoc). The only possibility therefore would be, to use this style to avoid additional checkstyle warnings:

```
 /**
   * Returns true if the ray generating this {@link Point} has hit an object.
   */
```